tests/testthat/test-observer.R

```diff
@@ -0,0 +1,47 @@
+test_that("can finalise observations automatically where similar", {
+  expect_equal(
+    observer_finalise_auto(list(list(a = 1:5), list(a = 6:10))),
+    list(a = cbind(1:5, 6:10)))
+  expect_equal(
+    observer_finalise_auto(list(list(a = 1:5, b = 1:3),
+                                list(a = 6:10, b = 4:6))),
+    list(a = cbind(1:5, 6:10),
+         b = cbind(1:3, 4:6)))
+})
+
+
+test_that("don't combine things that can't be stuck together", {
```

Comment:
something like that because it does combine the last one in the test
```suggestion
test_that("combine or don't combine observation correctly", {
```